src/main/java/dao/UserDao.java

```diff
@@ -0,0 +1,234 @@
+        double averageAge = 0;
+        int counter = 0;
+        String query = "Select id From Book Where title = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(query);
+        preparedStatement.setString(1, name);
+        ResultSet resultSet = preparedStatement.executeQuery();
+        resultSet.next();
+        id = resultSet.getInt(1);
+        preparedStatement.close();
+        resultSet.close();
+
+        query = "Select users.age from Users " +
+                "join journal on journal.user_id = users.id " +
+                "where journal.book_id = ?";
+        preparedStatement = connection.prepareStatement(query);
+        preparedStatement.setInt(1, id);
+        resultSet = preparedStatement.executeQuery();
+        while (resultSet.next()) {
+            averageAge += resultSet.getInt(1);
+            counter++;
+        }
+        return averageAge / counter;
+    }
+
+    public double getAverageAgeOfUsersByAuthor(String authorName, String authorSecondName,
+                                               String authorSurname) throws SQLException {
+        int id;
+        double averageAge = 0;
+        int counter = 0;
+        String query = "Select id From Author Where author_name = ? AND " +
+                "author_secondname = ? AND author_surname = ?";
+        PreparedStatement preparedStatement = connection.prepareStatement(query);
+        preparedStatement.setString(1, authorName);
+        preparedStatement.setString(2, authorSecondName);
+        preparedStatement.setString(3, authorSurname);
+        ResultSet resultSet = preparedStatement.executeQuery();
+        resultSet.next();
+        id = resultSet.getInt(1);
+        preparedStatement.close();
+        resultSet.close();
+
+        query = "Select users.age from Users " +
+                "join journal on journal.user_id = users.id " +
+                "join copy on journal.book_id = copy.book_id " +
+                "where copy.author_id = ?";
+        preparedStatement = connection.prepareStatement(query);
+        preparedStatement.setInt(1, id);
+        resultSet = preparedStatement.executeQuery();
+        while (resultSet.next()) {
+            averageAge += resultSet.getInt(1);
+            counter++;
+        }
+        return averageAge / counter;
+    }
+
+    public User logIn(String login, String password) {
+        String query = "Select * From Users " +
+                "Where login = ? AND user_password = ?";
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, login);
+            preparedStatement.setString(2, password);
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            User user = new UserMapper().rowMapper(resultSet).get(0);
+
+            return user;
+        } catch (SQLException e) {
+            return null;
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
why can't you just verify if collection is not empty instead of handling OutOfBound?